runtimes/huggingface/tests/test_runtime.py

```diff
@@ -29,10 +29,10 @@ async def test_infer_multiple(
 
     res = await runtime.predict(inference_request)
 
-    assert len(res.outputs) == 1
+    assert len(res.outputs) == 2
```

Comment:
That test case is running a `question-answering` task. the input length is 2,  So the task output is a list considered with two answers, like this `[{score: 1, answer: something}, {score: 1, answer: something}]`,  when encoding that data as a response,   RequestCodec `HuggingfaceRequestCodec` matched. ThenHuggingfaceRequestCodec will encode the data.  The HuggingfaceRequestCodec always considers that data is a batch output and always encodes elements in that data as JSON (why always JSON? I can't find a better way now. ). The output is like this 

```
{
    model_name: name,
    version: version,
    outputs: [
        {
           "name": "output_1",
           "data": ["{score: 1, answer: something}"]
        }, {
           "name": "output_2",
           "data": ["{score: 1, answer: something}"]
        }
    ]
}
```

Reply:
Ah right, in this case, it would fit better with the protocol to return a single output (e.g. named `output`), that contains two entries. When you look at the input, it's done a similar way (i.e. you will have a `question` input, with two elements, and a `context` input with two elements).

What codec is the one handling the encoding in this case? Would that be a big change over the current PR?

Reply:
In this case, RequestCodec is `HuggingfaceRequestCodec`. 

For the input side, HuggingfaceRequestCodec  will  find the input codec  in `HuggingfaceRequestCodec.InputCodecsWithPriority`, if not found, then found MLserver build-in InputCodec, if not found again, just `HuggingfaceRequestCodec.DefaultCodec` to encoding data; 

For the output side, the InputCodec `HuggingfaceJSONCodec` is always used;

That may be a big change, The default behavior changed, So should I keep this behavior like before, and keep hole prediction as a single output?  advice need here.

I noticed that `mlserver.codecs.utils.encode_inference_response` has a TODO, https://github.com/SeldonIO/MLServer/blob/5e8afc33c03e98f5fb226748dcfbe96d5fa753b5/mlserver/codecs/utils.py#L90-L95

If the codec could be overridden by metadata, This may be easy to deal with. But this may be another feature;

Reply:
I see... would there be any reason not to use the `HuggingfaceListJSONCodec` instead? That one seems to have the same logic as the runtime had previously, where multiple instances of the same output are treated as extra elements of the same `output` field.

Reply:
OK, I will change it to HuggingfaceListJSONCodec soon